src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -353,8 +355,16 @@ bool PerformInteractionRequest::ProcessVRResponse(
     return false;
   }
 
+  const hmi_apis::Common_Result::eType code =
+      static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asInt());
+  const bool is_result_successful =
+      CommandImpl::IsHMIResultSuccess(code, HmiInterfaces::HMI_INTERFACE_VR);
+
   if (IsInterfaceAwaited(HmiInterfaces::HMI_INTERFACE_UI) &&
-      InteractionMode::MANUAL_ONLY != interaction_mode_) {
+      InteractionMode::MANUAL_ONLY != interaction_mode_ &&
+      is_result_successful) {
```

Comment:
Hmmm, this also seems like it misses a few cases:

1. If an app sends `BOTH` mode and the HMI responds with `SUCCESS` to VR, `UI.ClosePopUp` should only be sent if `choiceID` is included, otherwise you would want to wait for a UI response.
2. If an app sends `VR_ONLY` mode and the HMI responds with `SUCCESS` to VR, `UI.ClosePopUp` should be sent regardless of whether or not `choiceID` is included, since the `choiceID` cannot come from the UI.